android/src/main/java/com/mattermost/networkclient/helpers/UploadProgressListener.kt

```diff
@@ -0,0 +1,27 @@
+package com.mattermost.networkclient.helpers
+
+import android.util.Log
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.modules.core.DeviceEventManagerModule.RCTDeviceEventEmitter
+import com.mattermost.networkclient.enums.APIClientEvents
+
+interface ProgressListenerInterface {
+    fun update(bytesRead: Double, contentLength: Double)
+    fun emitProgressEvent(progress: Double)
+}
+
+class ProgressListener(private val reactContext: ReactContext, private val taskId: String) : ProgressListenerInterface {
+
+    override fun emitProgressEvent(progress: Double) {
+        val params = Arguments.createMap()
+        params.putDouble("fractionCompleted", progress)
+        params.putString("taskId", "$taskId")
+        reactContext.getJSModule(RCTDeviceEventEmitter::class.java).emit(APIClientEvents.UPLOAD_PROGRESS.event, params)
+    }
+
+    override fun update(bytesRead: Double, contentLength: Double) {
+        val progress = bytesRead / contentLength;
```

Comment:
As we are passing the % value to JS, should we round the value? Wouldn't be better to send both values to JS and let the consumer use those values as they see fit? @migbot thoughts?

Reply:
I think the JS side is expecting a fraction between 1 and 0; and it can then workout what it wants to do with it

Reply:
What if I want to display the progress not in percentage? Every other lib sends both values in kb and you can calculate the % as you wish. We can send a double from 0.00 to 1.00 but then make sure the fraction only have two decimals